crates/iceberg/src/catalog/mod.rs

```diff
@@ -25,16 +25,16 @@ use crate::spec::{
 };
 use crate::table::Table;
 use crate::{Error, ErrorKind, Result};
-use async_trait::async_trait;
 use std::collections::HashMap;
 use std::mem::take;
 use std::ops::Deref;
 use typed_builder::TypedBuilder;
 use uuid::Uuid;
 
 /// The catalog API for Iceberg Rust.
-#[async_trait]
-pub trait Catalog: std::fmt::Debug {
+/// see https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html
+#[trait_variant::make(Catalog: Send)]
+pub trait LocalCatalog: std::fmt::Debug {
```

Comment:
Why we need this `LocalCatalog` variant? I think it would be enough to have a thread safe one.

Reply:
I think the better approach would be like `FileWriter` trait, which adds `Send` for each return type:

```rust
pub trait Catalog: Debug + Send {
  fn list_namespace(&self, parent: Option<&NamespeceIdent>) -> impl Future<Output = 
Result<Vec<NamespaceIdent>>> + Send;
}
```

Reply:
My goal was to minimize code changes as much as possible. There are 13 methods and 2 implementations of `Catalog`, while `trait_variant` is a one-line change. I personally prefer `async fn` over `impl Future` since it looks neater.  But it is true that `LocalCatalog` is never used.

Reply:
I'm hesitating adding this `LocalCatalog` trait. Also `async fn` seems much more concise to me. Maybe we should close this pr before finding a better approach?

Reply:
https://github.com/rust-lang/impl-trait-utils/pull/27

They are adding a `rewrite` feature to rewrite a trait instead of creating 2 variants. We can wait till it's done.

Reply:
Yeah, that's great! Let's wait for this.